corehq/apps/registration/views.py

```diff
@@ -77,15 +77,32 @@ def register_user(request, domain_type=None):
                     'REMOTE_ADDR': request.META.get('REMOTE_ADDR')
                 }
                 track_created_hq_account_on_hubspot.delay(new_user, request.COOKIES, meta)
-                return redirect(
-                    'registration_domain', domain_type=domain_type)
+                requested_domain = form.cleaned_data['hr_name']
+                if form.cleaned_data['create_domain']:
+                    org = None
+                    try:
+                        requested_domain = request_new_domain(
+                            request, form, org, new_user=True, domain_type=domain_type)
+                    except NameUnavailableException:
+                        context.update({
+                            'error_msg': _('Project name already taken - please try another'),
+                            'show_homepage_link': 1
+                        })
+                        return render(request, 'error.html', context)
```

Comment:
@amsagoff If the domain creation fails, you get taken to an error page that tells you the name is taken and then has a link to log in. Click the login link, and you go to the old "time to create your first project" page. I don't think this is perfect (ideally, you'd get the error displayed while you're still on the signup page), but it isn't horrific, and it should basically never happen (because of the "name-1", "name-2", etc. logic added in https://github.com/dimagi/commcare-hq/pull/7106).